qiskit/visualization/dag_visualization.py

```diff
@@ -225,38 +156,21 @@ def edge_attr_func(edge):
             e["label"] = label
             return e
 
-    dot_str = dag._multi_graph.to_dot(node_attr_func, edge_attr_func, graph_attrs)
-    dot = pydot.graph_from_dot_data(dot_str)[0]
-
     if filename:
         if "." not in filename:
             raise InvalidFileError("Parameter 'filename' must be in format 'name.extension'")
-        extension = filename.split(".")[-1]
-        if extension not in FILENAME_EXTENSIONS:
-            raise InvalidFileError(
-                "Filename extension must be one of: " + " ".join(FILENAME_EXTENSIONS)
-            )
-        dot.write(filename, format=extension)
-        return None
-    elif ("ipykernel" in sys.modules) and ("spyder" not in sys.modules):
-        _optionals.HAS_PIL.require_now("dag_drawer")
-        from PIL import Image
-
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            tmp_path = os.path.join(tmpdirname, "dag.png")
-            dot.write_png(tmp_path)
-            with Image.open(tmp_path) as test_image:
-                image = test_image.copy()
-            os.remove(tmp_path)
-            return image
+        return graphviz_draw(
+            dag._multi_graph,
+            node_attr_func,
+            edge_attr_func,
+            graph_attrs,
+            filename,
+            filename.split(".")[-1],
+        )
     else:
-        _optionals.HAS_PIL.require_now("dag_drawer")
-        from PIL import Image
-
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            tmp_path = os.path.join(tmpdirname, "dag.png")
-            dot.write_png(tmp_path)
-            image = Image.open(tmp_path)
-            image.show()
-            os.remove(tmp_path)
-            return None
+        return graphviz_draw(
```

Comment:
You should be able to do this without the if/else statement. If you just do something like:

```python3
image_type = None
if filename:
    if "." not in filename:
        raise InvalidFileError("Parameter 'filename' must be in format 'name.extension'")
    image_type = filename.split(".")[-1]
return graphviz_draw(
            dag._multi_graph,
            node_attr_func,
            edge_attr_func,
            graph_attrs,
            filename,
            filename.split(".")[-1],
)
```